### PR TITLE
story(issue-18): convert implement-go-text-file-library agent to skill

### LINKED ISSUE #18: story(implement-go-text-file-library): convert agent to skill
### Description

Replace the `implement-go-text-file-library` agent (181 lines) with a skill. The main motivator is `SPEC.md` context bloat: today the agent reads `SPEC.md` whole, partitions it into scratch files, and feeds those to phase subagents.

The skill version should never load the full `SPEC.md` into its own context. Instead it should identify section line ranges and pass `(file path, offset, limit)` to phase subagents — they read their slice directly via `Read`. Architecture patterns and testing conventions live in `references/` so `SKILL.md` stays slim.

Build using the `skill-creator` skill workflow.

### Acceptance Criteria

- [ ] Skill at `plugins/file-library/skills/implement-go-text-file-library/SKILL.md`
- [ ] `SKILL.md` under ~100 lines
- [ ] Phase orchestration logic in `SKILL.md`; pipeline patterns in `references/architecture.md`; testing conventions in `references/testing.md`
- [ ] Skill reads `SPEC.md` only enough to identify section line ranges, then delegates
- [ ] Phase subagents (tokenizer, parser, printer) read `SPEC.md` slices directly via `Read` with offset/limit
- [ ] Skill is built using the `skill-creator` skill workflow
- [ ] Old `agents/implement-go-text-file-library.md` deleted
- [ ] Plugin marketplace registers the skill

### Related Issues

#10

### PR DESCRIPTION
**Pull request by Zaba505**
## Summary

- Replace the 181-line `implement-go-text-file-library` agent with a slim skill at `plugins/file-library/skills/implement-go-text-file-library/`. `SKILL.md` is 88 lines and grep's `SPEC.md` for `## ` line ranges, handing each phase subagent a `(path, offset, limit)` slice rather than a full read — keeps the orchestrator's context lean for large specs.
- Architecture patterns (tokenizer/parser/printer action loops, the inner-action-loop rule for complex types, error patterns) live in `references/architecture.md`; testing conventions (collect helper, exact `Pos` values, parser-via-`Parse()`, round-trip pairing) live in `references/testing.md`.
- Built using the skill-creator workflow with 3 evals against a `kvr` text-format fixture (string record, nested block, comment trivia). Iteration-1: 97% with-skill vs 92% without; the skill earned cleaner test discipline and the inner-action-loop pattern at ~25% extra time/tokens.

Closes #18.

## Test plan

- [x] `SKILL.md` under 100 lines (88)
- [x] `references/architecture.md` and `references/testing.md` carry the load-bearing detail
- [x] Skill grep's SPEC.md for line ranges and never reads it whole
- [x] Phase subagents read SPEC.md slices via `Read(path, offset, limit)` per the SKILL.md instructions
- [x] Old `agents/implement-go-text-file-library.md` deleted
- [x] Iteration-1 eval workspace committed as the audit trail (3 evals × {with_skill, without_skill})
- [x] Plugin marketplace already registers the file-library plugin (skills auto-discovered)
- [ ] Reviewer should sanity-check the SKILL.md prose and the `references/architecture.md` framing — the eval loop validated discipline, not nuance

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/file-library/skills/implement-go-text-file-library/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+name: implement-go-text-file-library
+description: Implement features for Go text file library packages that follow a tokenizer/parser/printer pipeline. Use whenever the user wants to add token types, parser rules, AST nodes, or printer logic to a Go package built around `tokenizer.go`, `parser.go`, and `printer.go` — including phrases like "tokenize the X keyword", "parse the Y block", "format the Z node", "add support for comments", or "implement the spec section on records", even if the user doesn't say "text".
+---
+
+You are an orchestrator that adds features to an existing Go text file library package. You prepare context, then delegate each pipeline phase (tokenizer → parser → printer) to a focused subagent. You never read the full `SPEC.md` into your own context — large specs would crowd out orchestration. Instead, grep `SPEC.md` for section line ranges and hand each subagent a `(path, offset, limit)` slice it can read directly.
+
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/SKILL.md:7]: Addressed in 9a9102f. Added a **Re-run safety** item (#6) under "Before you start" that states: source files (`tokenizer.go`, `parser.go`, `printer.go`, and `_test.go` siblings) are amended via `Edit` and never recreated wholesale, so prior implementer work is preserved; the only files written from scratch are the two scratch summaries (`_context_tokens.md` / `_context_ast.md`), which are overwritten each phase and deleted in cleanup; if a previous run was interrupted and left either file behind, the orchestrator deletes them before launching Phase 1 so a stale partial summary cannot leak into the new run.

File: plugins/file-library/skills/implement-go-text-file-library/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+If `SPEC.md` is paired with already-chunked `tokens/<name>.md` or `grammar/<name>.md` files (the layout some `extract-text-spec` runs produce), pass those file paths verbatim to the relevant phase subagent — they're already chunked, so no slicing is needed.
+
+Before launching subagents, grep the slices for `> **Ambiguity:**` callouts and surface them to the user.
+
+## Phase order
+
+Run phases in order. Do not skip ahead. Each phase passes a small `_context_<phase>.md` summary forward.
+
+**If you have an `Agent` / `Task` tool available, spawn a subagent per phase** — it keeps the orchestrator's context lean. **If you don't, run each phase inline yourself**, in the same order, with the same slicing and the same `_context_<phase>.md` summaries between phases. The discipline (test-first, exact `Pos` values, inner action loop for complex types, round-trip tests) matters more than who executes the work.
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/SKILL.md]: Addressed in 9a9102f. Replaced both `_context_<phase>.md` template references in the **Phase order** section (the rule line and the inline-fallback paragraph) with the explicit `_context_tokens.md` / `_context_ast.md` filenames the downstream Phase 1/2 instructions already use. Kept the existing names rather than renaming to `_context_tokenizer.md` / `_context_parser.md` because they describe payload content (token types / AST types), which is more meaningful than the phase that produced them, and changing the names would have required touching three sites instead of one.

File: plugins/file-library/skills/implement-go-text-file-library/evals/fixtures/kvr/tokenizer.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		t.pos.Column++
+	}
+	return r, nil
+}
+
+// backup rewinds the last rune read by next.
+func (t *tokenizer) backup() {
+	_ = t.r.UnreadRune()
+	t.pos.Column--
+}
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/evals/fixtures/kvr/tokenizer.go]: Good catch — fixed in 9a9102f. Adopted the canonical `prevPos`/`hasPrev` snapshot pattern that the `new-go-text-file-library` scaffold already uses: `next()` now snapshots `t.prevPos = t.pos` and sets `t.hasPrev = true` before mutating `t.pos`, and `backup()` restores `t.pos = t.prevPos` (clearing `hasPrev`) only when `UnreadRune` succeeds — no column arithmetic, so it correctly handles the newline-boundary case. The iteration-1 eval-0 with-skill output had already had to invent its own fix (a `lastWasNewline` flag), confirming the fixture was the source. `go test -race ./...` passes against the fixture.

File: plugins/file-library/skills/implement-go-text-file-library/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+---
+name: implement-go-text-file-library
+description: Implement features for Go text file library packages that follow a tokenizer/parser/printer pipeline. Use whenever the user wants to add token types, parser rules, AST nodes, or printer logic to a Go package built around `tokenizer.go`, `parser.go`, and `printer.go` — including phrases like "tokenize the X keyword", "parse the Y block", "format the Z node", "add support for comments", or "implement the spec section on records", even if the user doesn't say "text".
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/SKILL.md]: [dfef815](https://github.com/z5labs/ai/pull/36/commits/dfef815c5f5dda62fc9edfd4e07e1efeb660e256)

File: plugins/file-library/skills/implement-go-text-file-library/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+Read `references/architecture.md` for the tokenizer/parser/printer patterns each subagent must follow (especially the **inner action loop** rule for complex types — flat for-loops with switches do not scale and must be rejected in review).
+Read `references/testing.md` for text-specific test conventions before launching any subagent.
+
+## Before you start
+
+1. Read the package's `CLAUDE.md` (if present) and the repo-root `CLAUDE.md` for project conventions and license-header style.
+2. List the package: confirm `tokenizer.go`, `parser.go`, `printer.go`, and their `_test.go` siblings exist. If they don't, the user wants the `new-go-text-file-library` scaffold first — say so and stop.
+3. Check for `<package>/SPEC.md`. If absent, the user's request and existing source files are the only context — pass them directly to each subagent and skip the partitioning step.
+4. Identify scope: which token types, AST nodes, parser rules, and printer rules will change.
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/SKILL.md:28]: [dfef815](https://github.com/z5labs/ai/pull/36/commits/dfef815c5f5dda62fc9edfd4e07e1efeb660e256)

File: plugins/file-library/skills/implement-go-text-file-library/references/architecture.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+```go
+func tokenize(t *tokenizer, yield func(Token, error) bool) tokenizerAction {
+    r, err := t.next()
+    if err == io.EOF { return nil }
+    if err != nil    { return yieldErr(err) }
+    switch {
+    case unicode.IsSpace(r):  return skipWhitespace
+    case r == '#':            return tokenizeComment(t.pos)  // closure captures start pos
+    case r == '"':            return tokenizeString(t.pos)
+    case unicode.IsDigit(r):  t.backup(); return tokenizeNumber
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/references/architecture.md:50]: [4898c0f](https://github.com/z5labs/ai/pull/36/commits/4898c0fb29e7124312d7c730a18d055e8d664d6a)

File: plugins/file-library/skills/implement-go-text-file-library/references/testing.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+```go
+// WRONG — hand-constructs the AST, hides parser regressions
+want := &File{Records: []Record{{Type: "string", Value: "hello"}}}
+
+// RIGHT — uses Parse on a canonical source string as the expectation source
+want, err := Parse(strings.NewReader(`record string = "hello"`))
+require.NoError(t, err)
+got, err := Parse(strings.NewReader(testInput))
+require.NoError(t, err)
+require.Equal(t, want, got)
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/references/testing.md:80]: [4898c0f](https://github.com/z5labs/ai/pull/36/commits/4898c0fb29e7124312d7c730a18d055e8d664d6a)

File: plugins/file-library/skills/implement-go-text-file-library/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+- **`<package>/SPEC.md`** (optional) — when present, sliced by line range per phase; when absent, the user's request plus existing source files are the only context.
+- **`<package>/tokens/*.md`, `<package>/grammar/*.md`** (optional) — pre-chunked spec files produced by `extract-text-spec`. Passed to subagents verbatim, no slicing.
+
+## Outputs
+
+- **Edits** to `<package>/tokenizer.go`, `<package>/tokenizer_test.go`, `<package>/parser.go`, `<package>/parser_test.go`, `<package>/printer.go`, `<package>/printer_test.go` — amended via `Edit`, never recreated wholesale, so prior implementer work is preserved.
+- **Scratch files** `<package>/_context_tokens.md` (after Phase 1) and `<package>/_context_ast.md` (after Phase 2) — overwritten each run, deleted in Cleanup. If a previous run was interrupted and left either file behind, delete them before launching Phase 1 so a stale partial summary cannot leak into the new run.
+- **Side effect**: runs `go test -race ./...` between phases to verify each phase before launching the next.
+
```
[the remark this answers is not in the document]
Comment (Zaba505) [reply plugins/file-library/skills/implement-go-text-file-library/SKILL.md:22]: [4b26cc1](https://github.com/z5labs/ai/pull/36/commits/4b26cc12d1c8d76b76e4b3a0d65fc9fcaaa9c45c)
